test/erlang_ls_fake_stdio.erl

```diff
@@ -0,0 +1,111 @@
+-module(erlang_ls_fake_stdio).
```

Comment:
This file was created to emulated the communication between client and server through stdio.

Reply:
Can you expand? Why can't we use the normal server/client for the tests (I guess intercepting all other output)?

Reply:
Because the client needs a way to *write to the server's stdin* and a way to *read from the server's stdout*. This can't be achieved in a simple way that I know of.

Reply:
Could a file be used as a IO device for the tests?

Reply:
I thought about that, but I couldn't think of a way to make it work with files. We would need two files `A` and `B`, and some code that would use:
- `A` as stdout for the server and `B` as stdin.
- `B` as stdout for the client and `A` as stdin.
This code would need to be a process that implements an IO server and uses `A` and `B` as described above.

Reply:
Let's live with this for now.